packages/smooth_app/lib/pages/preferences/user_preferences_account.dart

```diff
@@ -361,12 +366,38 @@ class _UserPreferencesPageState extends State<UserPreferencesSection> {
     return Column(children: result);
   }
 
+  Future<int?> _getMyCount(
+    final UserProductSearchType type,
+  ) async {
+    final UserProductSearchQueryConfiguration configuration =
+        UserProductSearchQueryConfiguration(
+      type: type,
+      userId: OpenFoodAPIConfiguration.globalUser!.userId,
+      pageSize: 1,
+      language: ProductQuery.getLanguage(),
+      fields: [],
+    );
+
+    final SearchResult result;
+    try {
+      result = await OpenFoodAPIClient.searchProducts(
```

Comment:
You could directly return the count here

Reply:
giving error in type casting

Reply:
@g123k @prathamsoni11 Let's say that we could do everything that works in a `try`, and return `null` in case of `Exception`:
```dart
try {
  final SearchResult result = await OpenFoodAPIClient.searchProducts(
    OpenFoodAPIConfiguration.globalUser,
    configuration,
    queryType: OpenFoodAPIConfiguration.globalQueryType,
  );
  return result.count;
} catch (e) {
  return null;
}
```

Reply:
ok done👍🏻

Reply:
That said, I don't know if it is what @g123k had in mind.
Anyway, now your code passes the tests.